src/realm/group.cpp

```diff
@@ -419,6 +419,7 @@ int Group::read_only_version_check(SlabAlloc& alloc, ref_type top_ref, const std
         case 0:
             file_format_ok = (top_ref == 0);
             break;
+        case 24:
```

Comment:
why this magic number here?

Reply:
This is the version(s) we can open in read-only mode

Reply:
OK, got it, because v24 will be eventually rewritten in v25 after compression!

Reply:
Because we can read v24 without upgrade. If it is in read-only mode, we will not make any commits and therefore no compression will happen.